salt/utils/dictupdate.py

```diff
@@ -50,7 +54,7 @@ def update(dest, upd, recursive_update=True, merge_lists=False):
             elif isinstance(dest_subkey, list) \
                      and isinstance(val, list):
                 if merge_lists:
-                    dest[key] = dest.get(key, []) + val
+                    dest[key] = sorted((set(dest_subkey + val)))
```

Comment:
I don't agree with sorting the merged result. There doesn't appear to be any reason to do so, and it represents a significant change to how this has operated in the past. We can just remove duplicates using a simple list comprehension, there's no need to use sets and sort the result:

```python
new_list = dest.get(key, [])
new_list.extend([x for x in val if x not in new_list])
dest[key] = new_list
```